File: src/Offering/OfferingTest/DiscountTest.cs
using Offering.Models;

namespace OfferingTest;

public class DiscountTest
{
    [Fact]
    public void Discount_Is_Instantiated_Correctly()
    {
        var dateStart = DateTime.Now;
        var dateEnd = DateTime.MaxValue;
        var discount = Discount.Create(10, dateStart, dateEnd);
        
        Assert.Equal(10, discount.DiscountRate);
        Assert.Equal(dateStart, discount.StartDate);
        Assert.Equal(dateEnd, discount.EndDate);
    }
}

File: src/OfferSB/src/test/java/com/example/offersb/test/UnitTests/DiscountTest.java
package com.example.offersb.test.UnitTests;

import com.example.offersb.models.Discount;
import org.junit.jupiter.api.Test;

import java.time.LocalDateTime;

public class DiscountTest {

    @Test
    public void testDiscount_Is_Instantiated_Correctly() {
        var dateStart = LocalDateTime.now();
        var dateEnd = LocalDateTime.MAX;
        var discount = Discount.create(10, dateStart, dateEnd);

        assert discount.getDiscountRate() == 10;
        assert discount.getStartDate() == dateStart;
        assert discount.getEndDate() == dateEnd;
    }
}
